### ISSUE TITLE
Autocomplete is incompatible with qualified partial imports

### ISSUE THREAD

**Issue by serin-delaunay** [nwolverson/atom-ide-purescript#125]
I have the following Main.purs:

```
module Main where

import Prelude
import Control.Monad.Eff (Eff)
import Control.Monad.Eff.Console (CONSOLE, logShow) as C

main :: forall e. Eff (console :: C.CONSOLE | e) Unit
main = do
  C.logShow 4
```

It builds with no issues. If I add a new line (after building) with `notmain = do C.lo` then autocomplete fails to suggest `C.logShow`.
On the other hand, if I add a line with `notmain = do lo` then autocomplete suggests `logShow`, which does not exist in this context.

**Comment by kritzcreek** [comment]
It looks like the plugin's import parsing is breaking down here. Why would you explicitly list the imports for a qualified import though? If you just go with `import Control.Monad.Eff.Console as C` and type `C.` you will get all the right completions. The reason for why identifiers that are not imported yet pop up in the completion list is related to #126. If we don't want to manually edit import statements, we need to be able to autocomplete not yet imported functions.

**Comment by nwolverson** [comment]
You would explicitly list imports for a qualified import when importing multiple modules with the same qualifier - this is recommended for some things and will otherwise trigger import warnings.

Will take a look at this when I can, my expectation is that for explicit qualified imports, everything with that qualifier should be suggested or just the imported things, according to the settings.

**Comment by nwolverson** [comment]
This is fixed as of 0c4e01ab8327ccac53716017cec6dfed65109a58 along with updates in the current compiler master, in that explicit/hiding imports with qualifier will suggest completions with the qualifier. The following caveats apply
- Suggestions are never filtered for explicit/hiding identifiers, which isn't quite right when _not_ adding imports on completion
- We never add imports on completion for qualified explicit imports (this would require compiler changes and I don't feel it's worth it unless demand shows otherwise - you can remove the explicit list and apply the auto-fix suggestion afterwards)

**Comment by nwolverson** [comment]
Fixed in 0.10
